tests/data/test_types.py

```diff
@@ -434,3 +456,65 @@ def test_parse_path():
     # Parse optional path. Works the same way but passes None down the line.
     assert util.parse_optional_path(None) is None
     assert util.parse_optional_path("/foo/bar") == util.parse_path("/foo/bar")
+
+
+def test_dict_obs():
+    A = np.random.rand(3, 4)
+    B = np.random.rand(3, 7, 1)
+    C = np.random.rand(4)
+
+    ab = types.DictObs({"a": A, "b": B})
+    abc = types.DictObs({"a": A, "b": B, "c": C})
+
+    # len
+    assert len(ab) == 3
+    with pytest.raises(RuntimeError):
+        len(abc)
+    with pytest.raises(RuntimeError):
+        len(types.DictObs({}))
+
+    assert abc.dict_len == 3
+
+    # slicing
+    np.testing.assert_equal(abc[0].get("a"), A[0])
+    np.testing.assert_equal(abc[0].get("c"), np.array(C[0]))
+    np.testing.assert_equal(abc[0:2].get("a"), np.array(A[0:2]))
+    np.testing.assert_equal(ab[:, 0].get("a"), np.array(A[:, 0]))
+    with pytest.raises(IndexError):
+        abc[:, 0]
+
+    # iter
+    for i, a_row in enumerate(A):
+        np.testing.assert_equal(a_row, ab[i].get("a"))
+    assert ab[0] == next(iter(ab))
+
+    # eq
+    assert abc == types.DictObs({"a": A, "b": B, "c": C})
```

Comment:
```suggestion
    assert abc == types.DictObs({"a": A, "b": B, "c": C})
    assert abc == types.DictObs({"a": np.array(A), "b": np.array(B), "c": np.array(C)})
```
In case just testing NumPy array identity rather than equality